retrieval/scrape.go

```diff
@@ -297,6 +307,7 @@ func (s *targetScraper) scrape(ctx context.Context, ts time.Time) (model.Samples
 	}
 	req.Header.Add("Accept", acceptHeader)
 	req.Header.Set("User-Agent", userAgentHeader)
+	req.Header.Set("Scrape-Timeout-Seconds", fmt.Sprintf("%f", s.timeout.Seconds()))
```

Comment:
I know this header name was mentioned in https://github.com/prometheus/prometheus/issues/2508, but shouldn't we use `X-Prometheus-Scrape-Timeout-Seconds` instead? We prefix all our other headers with `X-Prometheus-`:

- `X-Prometheus-API-Version` (scrape protocol version)
- `X-Prometheus-Remote-Write-Version`
- `X-Prometheus-Remote-Read-Version`

Although I just found https://tools.ietf.org/html/rfc6648, which says that `X-` is kinda not in fashion anymore. Still, we should be consistent (I'd still include the `X-` for now) and at least mention Prometheus?